services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/service/CommandProcessorService.java

```diff
@@ -0,0 +1,270 @@
+        int count;
+        synchronized (rejectedQueue) {
+            while (!rejectedQueue.isEmpty()) {
+                Runnable entry = rejectedQueue.getFirst();
+                if (queue.offer(entry)) {
+                    rejectedQueue.removeFirst();
+                    continue;
+                }
+
+                break;
+            }
+            count = rejectedQueue.size();
+        }
+
+        reportQueueStatus(count);
+    }
+
+    private void verifyPendingStatus() {
+        LinkedList<ProcessorTask> checkList = rotatePendingCommands();
+        if (checkList.size() == 0) {
+            return;
+        }
+
+        try {
+            CommandContext context = commandContextFactory.produce();
+            VerifyBatch verifyBatch = new VerifyBatch(this, checkList);
+            PendingCommandSubmitter checkCommands = new PendingCommandSubmitter(context, verifyBatch);
+            processLazy(checkCommands);
+        } catch (Throwable e) {
+            synchronized (this) {
+                tasks.addAll(checkList);
+            }
+            throw e;
+        }
+    }
+
+    private void reportQueueStatus(int rejectedQueueSize) {
+        if (0 < rejectedQueueSize) {
+            long now = System.currentTimeMillis();
+            if (lastRejectCountReportedAt + REJECTED_REPORT_INTERVAL < now) {
+                lastRejectCountReportedAt = now;
+
+                String message = String.format(
+                        "Rejected commands queue size: %d (workers: %d / %d)",
+                        rejectedQueueSize, executor.getActiveCount(), executor.getPoolSize());
+                if (rejectedQueueSize < REJECTED_ERROR_LIMIT) {
+                    log.warn(message);
+                } else {
+                    log.error(message);
+                }
+            }
+        } else if (0 < lastRejectCountReportedAt) {
+            log.warn("All rejected command have been submitted into executor");
+            lastRejectCountReportedAt = 0;
+        }
+    }
+
+    private synchronized LinkedList<ProcessorTask> rotatePendingCommands() {
+        LinkedList<ProcessorTask> current = tasks;
+        tasks = new LinkedList<>();
+        return current;
+    }
+
+    private void scheduleFutureCheckTrigger(ScheduledExecutorService scheduler) {
+        scheduler.scheduleAtFixedRate(
+                this::timerTrigger,
+                FUTURE_COMPLETE_CHECK_INTERVAL, FUTURE_COMPLETE_CHECK_INTERVAL, TimeUnit.MILLISECONDS);
+    }
+
+    public static class VerifyBatch implements AutoCloseable {
+        private CommandProcessorService commandProcessor;
+        private final LinkedList<ProcessorTask> tasksBatch;
+
+        VerifyBatch(CommandProcessorService commandProcessor, LinkedList<ProcessorTask> tasksBatch) {
+            this.commandProcessor = commandProcessor;
+            this.tasksBatch = tasksBatch;
+        }
+
+        @Override
+        public void close() {
```

Comment:
Why the method called "close", but allows to be invoked multiple times? Please, reconsider the name.